components/ILIAS/Calendar/classes/class.ilDatePresentation.php

```diff
@@ -154,59 +171,101 @@ public static function formatDate(ilDateTime $date, bool $a_skip_day = false, bo
-                $sep = "";
-                if (!$a_skip_starting_day) {
-                    $date_str = self::formatDate(
-                        new ilDate($start->get(IL_CAL_DATE, '', $ilUser->getTimeZone()), IL_CAL_DATE)
-                    );
-                    $sep = ", ";
-                }
+                return self::formatDate(
+                    $start,
+                    false,
+                    false,
+                    false,
+                    $user
+                );
+            }
+            $date_str = '';
+            $sep = '';
+            if (!$a_skip_starting_day) {
+                $date_str = self::formatDate(
+                    new ilDate(
+                        $start->get(IL_CAL_DATE, '', $user->getTimeZone()),
+                        IL_CAL_DATE
+                    ),
+                    false,
+                    false,
+                    false,
+                    $user
+                );
+                $sep = ', ';
+            }
 
-                // $start == $end
-                if (ilDateTime::_equals($start, $end)) {
-                    switch ($ilUser->getTimeFormat()) {
-                        case ilCalendarSettings::TIME_FORMAT_24:
-                            return $date_str . $sep . $start->get(IL_CAL_FKT_DATE, 'H:i', $ilUser->getTimeZone());
-
-                        case ilCalendarSettings::TIME_FORMAT_12:
-                            return $date_str . $sep . $start->get(IL_CAL_FKT_DATE, 'h:i a', $ilUser->getTimeZone());
-                    }
-                } else {
-                    switch ($ilUser->getTimeFormat()) {
-                        case ilCalendarSettings::TIME_FORMAT_24:
-                            return $date_str . $sep . $start->get(
+            if (ilDateTime::_equals($start, $end)) {
+                switch ($user->getTimeFormat()) {
+                    case ilCalendarSettings::TIME_FORMAT_24:
+                        return $date_str . $sep .
+                            $start->get(
                                 IL_CAL_FKT_DATE,
                                 'H:i',
-                                $ilUser->getTimeZone()
+                                $user->getTimeZone()
+                            );
+                    case ilCalendarSettings::TIME_FORMAT_12:
+                        return $date_str . $sep .
+                            $start->get(
+                                IL_CAL_FKT_DATE,
+                                'h:i a',
+                                $user->getTimeZone()
+                            );
+                }
+            } else {
+                switch ($user->getTimeFormat()) {
+                    case ilCalendarSettings::TIME_FORMAT_24:
+                        return $date_str . $sep .
+                            $start->get(
+                                IL_CAL_FKT_DATE,
+                                'H:i',
+                                $user->getTimeZone()
```

Comment:
@fhelfer Please have a look at these kind of changes. This costs really lot of time to check, if this code "has no changes at all". Please, if you provide bug fix PRs, do not include so many code style improvements.